src/server/src/v1/startup/application/services/startupService.factory.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common'
+import { DarwinStartupService } from './darwinStartup.service'
+import { UnsupportedStartupService } from './unsupportedStartup.service'
+
+@Injectable()
+export class StartupServiceFactory {
+  constructor(
+    private readonly darwinStartupService: DarwinStartupService,
+    private readonly unsupportedStartupService: UnsupportedStartupService,
+  ) {}
+
+  getStartupService(): CodeClimbers.StartupService {
+    const os = process.platform
+    switch (os) {
+      case 'darwin':
+        return this.darwinStartupService
+      default:
```

Comment:
to add more platforms to support , add additional case statements and new services that implement CodeClimbers.StartupService